### PR TITLE
Clean up javadoc and reformat code

### PR DESCRIPTION
**Pull request by wallner**
This commit doesn't change the behaviour of the code at all, it just
reformats it and fixes up the javadoc.

### CODE REVIEW

File: src/main/java/org/osiam/resources/scim/Address.java [hunk 1]
```diff
@@ -317,7 +317,7 @@ public Builder setPostalCode(String postalCode) {
         }
 
         /**
-         * Sets the label indicating the attribute's function (See {@link MultiValuedAttribute#getType()}).
+         * Sets the label indicating the attribute's function (See {@link MultiValuedAttribute#getValue()}).
```
Comment (tkrille) [inline src/main/java/org/osiam/resources/scim/Address.java]: this should remain `getType()`.

File: src/main/java/org/osiam/resources/scim/Email.java [hunk 2]
```diff
@@ -186,7 +184,7 @@ public Builder setValue(String value) {
         }
 
         /**
-         * Sets the label indicating the attribute's function (See {@link MultiValuedAttribute#getType()}).
+         * Sets the label indicating the attribute's function (See {@link MultiValuedAttribute#getValue()}).
```
Comment (tkrille) [inline src/main/java/org/osiam/resources/scim/Email.java]: this should remain `getType()`.

File: src/main/java/org/osiam/resources/scim/Entitlement.java [hunk 3]
```diff
@@ -165,10 +162,9 @@ public Builder setValue(String value) {
         }
 
         /**
-         * Sets the label indicating the attribute's function (See {@link MultiValuedAttribute#getType()}).
-         * 
-         * @param type
-         *        the type of the attribute
+         * Sets the label indicating the attribute's function (See {@link MultiValuedAttribute#getValue()}}).
```
Comment (tkrille) [inline src/main/java/org/osiam/resources/scim/Entitlement.java]: this should remain `getType()`.

File: src/main/java/org/osiam/resources/scim/GroupRef.java [hunk 4]
```diff
@@ -139,8 +137,8 @@ public Builder setValue(String value) {
         }
 
         /**
-         * Sets the label indicating the attribute's function (See {@link MultiValuedAttribute#getType()}).
-         * 
+         * Sets the label indicating the attribute's function (See {@link MultiValuedAttribute#getValue()}).
```
Comment (tkrille) [inline src/main/java/org/osiam/resources/scim/GroupRef.java]: this should remain `getType()`.

File: src/main/java/org/osiam/resources/scim/Im.java [hunk 5]
```diff
@@ -166,10 +162,9 @@ public Builder setValue(String value) {
         }
 
         /**
-         * Sets the label indicating the attribute's function (See {@link MultiValuedAttribute#getType()}).
-         * 
-         * @param type
-         *        the type of the attribute
+         * Sets the label indicating the attribute's function (See {@link MultiValuedAttribute#getValue()}).
```
Comment (tkrille) [inline src/main/java/org/osiam/resources/scim/Im.java]: this should remain `getType()`.

File: src/main/java/org/osiam/resources/scim/PhoneNumber.java [hunk 6]
```diff
@@ -166,8 +166,8 @@ public Builder setValue(String value) {
         }
 
         /**
-         * Sets the label indicating the attribute's function (See {@link MultiValuedAttribute#getType()}).
-         * 
+         * Sets the label indicating the attribute's function (See {@link MultiValuedAttribute#getValue()}).
```
Comment (tkrille) [inline src/main/java/org/osiam/resources/scim/PhoneNumber.java]: this should remain `getType()`.

File: src/main/java/org/osiam/resources/scim/Photo.java [hunk 7]
```diff
@@ -237,10 +229,9 @@ public Builder setValue(ImageDataURI image) {
         }
 
         /**
-         * Sets the label indicating the attribute's function (See {@link MultiValuedAttribute#getType()}).
-         * 
-         * @param type
-         *        the type of the attribute
+         * Sets the label indicating the attribute's function (See {@link MultiValuedAttribute#getValue()}).
```
Comment (tkrille) [inline src/main/java/org/osiam/resources/scim/Photo.java]: this should remain `getType()`.

File: src/main/java/org/osiam/resources/scim/Role.java [hunk 8]
```diff
@@ -138,10 +133,9 @@ public Builder setPrimary(boolean primary) {
         }
 
         /**
-         * Sets the label indicating the attribute's function (See {@link MultiValuedAttribute#getType()}).
-         * 
-         * @param type
-         *        the type of the attribute
+         * Sets the label indicating the attribute's function (See {@link MultiValuedAttribute#getValue()}).
```
Comment (tkrille) [inline src/main/java/org/osiam/resources/scim/Role.java]: this should remain `getType()`.

File: src/main/java/org/osiam/resources/scim/SCIMSearchResult.java [hunk 9]
```diff
@@ -45,7 +44,7 @@
     private long totalResults;
     private long itemsPerPage;
     private long startIndex;
-    private Set<String> schemas = new HashSet<>(Arrays.asList(SCHEMA));
+    private Set<String> schemas = new HashSet<>(Collections.singletonList(SCHEMA));
```
Comment (tkrille) [inline src/main/java/org/osiam/resources/scim/SCIMSearchResult.java]: This changes the semantics of the code. `Arrays.asList(SCHEMA)` returns a mutable list, `Collections.singletonList(SCHEMA)` on the other hand returns an immutable list.

File: src/main/java/org/osiam/resources/scim/X509Certificate.java [hunk 10]
```diff
@@ -138,10 +133,9 @@ public Builder setPrimary(boolean primary) {
         }
 
         /**
-         * Sets the label indicating the attribute's function (See {@link MultiValuedAttribute#getType()}).
-         * 
-         * @param type
-         *        the type of the attribute
+         * Sets the label indicating the attribute's function (See {@link MultiValuedAttribute#getValue()}.
```
Comment (tkrille) [inline src/main/java/org/osiam/resources/scim/X509Certificate.java]: this should remain `getType()`.

File: src/main/java/org/osiam/resources/scim/Address.java (hunk 1 above)
Comment (wallner) [inline src/main/java/org/osiam/resources/scim/Address.java]: I see what I did there. Stupid me.

**Comment by tkrille** [comment]
I'd suggest, while we at it, to remove the `NOSONAR` comments. They're nothing but line noise and we can achieve the same effect via the Sonar's UI. I've also found some inconsistencies in styling, e.g. https://github.com/osiam/scim-schema/pull/162/files#diff-327e4dcbb0d9f399e929971daa89bd58R189, https://github.com/osiam/scim-schema/pull/162/files#diff-281e298fa8c4f1a13192aa2358f7553bR142, https://github.com/osiam/scim-schema/pull/162/files#diff-ddf656ad1da358dc789c87efc5873260R141, and maybe more. Please check the changed files again for a consistent style.

File: src/main/java/org/osiam/resources/scim/SCIMSearchResult.java (hunk 9 above)
Comment (wallner) [inline src/main/java/org/osiam/resources/scim/SCIMSearchResult.java]: Well, yes. You are right. I'll move that to a different PR

**Comment by wallner** [comment]
Ok. Done.

**Comment by tkrille** [comment]
Thank you!

**Pull request merged**